src/Microsoft.AspNetCore.Http.Connections.Client/Internal/PipeReaderFactory.cs

```diff
@@ -30,9 +30,7 @@ public static PipeReader CreateFromStream(PipeOptions options, Stream stream, Ca
             {
                 try
                 {
-                    // REVIEW: Should we use the default buffer size here?
-                    // 81920 is the default bufferSize, there is no stream.CopyToAsync overload that takes only a cancellationToken
-                    await stream.CopyToAsync(new PipeWriterStream(pipe.Writer), bufferSize: 81920, cancellationToken);
+                    await stream.CopyToAsync(new PipeWriterStream(pipe.Writer), bufferSize: 4096, cancellationToken);
```

Comment:
What about we write a custom method for copying from a stream directly to a pipe writer? Can avoid allocating the `PipeWriterStream` and maybe allocating the buffer. Can be done in the future, but add a comment.

Reply:
Or a new issue

Reply:
Not a pipe writer a pipe reader. See https://github.com/dotnet/corefx/issues/27246
. Pretty out of scope right now